crepecake/src/main/java/com/google/cloud/tools/crepecake/json/JsonParser.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.json;
+
+import com.google.cloud.tools.crepecake.image.Digest;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/** Interface to a custom {@link Gson} JSON parser. */
+public class JsonParser {
```

Comment:
Kinda makes it hard to use this in testing if everything is static? Would it make sense to singleton this or something in that style? Or am I miss-guessing how this will be used?

Reply:
Yea, this is sorta a singleton, except that it is a singleton of a Gson class.